ansible/roles/probe-services/templates/tor_targets.json

```diff
@@ -300,5 +300,31 @@
       ]
     },
     "protocol": "obfs4"
+  },
+  "80390900eecf5e7820083775d66c8058d4cc2098840ec42cf1e2046f50552c9b": {
```

Comment:
This is a duplicate key in the JSON object, I guess because the bridge is reachable on both ipv4 and ipv6 with the same fingerprint.

I think we need to think of how we go about testing ipv6 bridges as it's something we did not consider up until now.

An option could be of adding an additional key for the ipv6 address (`address_ipv6`?) and handle it in the testing logic.

@NullHypothesis given a bridge which is reachable on ipv6 and ipv4 can other parameters of the config (ex. the shared secret) vary depending on if you reach it via v4 or v6?

Reply:
> @NullHypothesis given a bridge which is reachable on ipv6 and ipv4 can other parameters of the config (ex. the shared secret) vary depending on if you reach it via v4 or v6?

In this particular case that cannot happen because this bridge runs a single Tor process that spawned a single obfs4proxy process. Other parameters can vary however if somebody set up an IPv4/IPv6 bridge by running two separate Tor instances which spawn two separate obfs4proxy instances – but I believe you weren't asking about that.

Reply:
> I think we need to think of how we go about testing ipv6 bridges as it's something we did not consider up until now.

Append `/ipv4` or `/ipv6` to the fingerprint and then hash that? (Or append `/ipv4` or `/ipv6` to the hash, which maybe is even more clear?)

Reply:
> > I think we need to think of how we go about testing ipv6 bridges as it's something we did not consider up until now.
> 
> Append `/ipv4` or `/ipv6` to the fingerprint and then hash that? (Or append `/ipv4` or `/ipv6` to the hash, which maybe is even more clear?)

I'm happy to do that but why not make the key be addr:port as it is done for the directory authorities?

Reply:
> I'm happy to do that but why not make the key be addr:port as it is done for the directory authorities?

The reason for that is that we would like to eventually support testing also private bridges which come from bridge-db and if we are using the IP and port as key, we would be disclosing the address of the bridge in the result.

As a matter of fact, you can use whatever you prefer as a key for the testing target, we figured using a hash of the fingerprint was a sane approach that would work well also in the case of private bridges.

If you would prefer we use a different key, we can do that too.

Reply:
That makes sense, thanks for elaborating! I'll revise my pull request.

Reply:
In 13d7c81, I derived a bridge's key from the concatenation of its IP:port tuple and its fingerprint. This should be future-proof and even works in a world where a Tor bridge exposes multiple IPv4/IPv6 endpoints using a single identity.